admin/autoupdater/src/manifest.c

```diff
@@ -47,6 +49,24 @@ void clear_manifest(struct manifest *m) {
 }
 
 
+void dump_manifest(const char *info_path, struct manifest *m) {
+	int fd = open(info_path, O_WRONLY|O_CREAT, 0600);
```

Comment:
* How about writing the data to stdout instead?
* Using fopen/fprintf (or printf in the case of stdout) would simlify this code a a lot (and is generally more efficient than using open/write, as it doesn't require a syscall for each function call)